fs.go

```diff
@@ -0,0 +1,183 @@
+package registry
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+)
+
+const (
+	registryDir       = "registry"
+	registryEntityDir = "entity"
+
+	placeholderFilename = ".placeholder"
+	statementExt        = ".json"
+)
+
+// MutableProvider is a mutable registry provider interface.
+type MutableProvider interface {
+	Provider
+
+	// BaseDir returns the base registry directory (when available).
+	BaseDir() string
+
+	// Init initializes a new registry in the local filesystem.
+	Init() error
+
+	// Verify verifies the integrity of the whole registry.
+	Verify() error
```

Comment:
It might make more sense moving `Verify()` in the `Provider` interface, as it might be useful also to read-only providers for verifying the whole registry.